core/src/main/kotlin/gui/EndGameScreen.kt

```diff
@@ -0,0 +1,124 @@
+package gui
+
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.Screen
+import com.badlogic.gdx.graphics.GL20
+import com.badlogic.gdx.graphics.Texture
+import com.badlogic.gdx.graphics.g2d.Sprite
+import com.badlogic.gdx.graphics.g2d.SpriteBatch
+import com.badlogic.gdx.graphics.g2d.TextureRegion
+import com.badlogic.gdx.scenes.scene2d.InputEvent
+import com.badlogic.gdx.scenes.scene2d.Stage
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
+import com.badlogic.gdx.utils.viewport.ScreenViewport
+import core.GameState
+import java.util.logging.Level
+import java.util.logging.Logger
+import kotlin.system.exitProcess
+
+class EndGameScreen(val game: Game, private val state: GameState) : Screen {
+
+    init {
+        logger.log(Level.INFO, "EndGameScreen's constructor")
+    }
+
+    lateinit var stage: Stage
+    lateinit var back: Texture
+    lateinit var batch: SpriteBatch
+    lateinit var backSprite: Sprite
+
+    companion object {
+        private val logger = Logger.getLogger(this::class.simpleName)
+    }
+
+    private val menuButton = Button(Texture(GUIConstants.menuButtonUpPath),
+            Texture(GUIConstants.menuButtonDownPath),
+            GUIConstants.menuButtonX, GUIConstants.menuButtonY,
+            GUIConstants.menuButtonWidth, GUIConstants.menuButtonHeight)
+
+    private val exitButton = Button(Texture(GUIConstants.exitButtonUpPath),
+            Texture(GUIConstants.exitButtonDownPath),
+            GUIConstants.exitButtonXE, GUIConstants.exitButtonYE,
+            GUIConstants.exitButtonWidthSE, GUIConstants.exitButtonHeightSE)
+
+
+    private fun setBackGround() {
+        back = Texture(GUIConstants.backImagePath)
+        back.setFilter(Texture.TextureFilter.Linear, Texture.TextureFilter.Linear)
+        backSprite = Sprite(back)
+        backSprite.setBounds(GUIConstants.backPosX, GUIConstants.backPosY,
+                GUIConstants.backWidth, GUIConstants.backHeight)
+
+    }
+
+    private fun setButtons() {
+        menuButton.addListener(object : ClickListener() {
```

Comment:
This listener overload should be also used here

Reply:
Fixed